src/Umbraco.Core/Persistence/Repositories/RepositoryBase.cs

```diff
@@ -258,12 +258,20 @@ public IEnumerable<TEntity> GetAll(params TId[] ids)
                 //.Where(x => Equals(x, default(TId)) == false)
                 .ToArray();
 
-            if (ids.Length > 2000)
+            // can't query more than 2000 ids at a time... but if someone is really querying 2000+ entities,
+            // the additional overhead of fetching them in groups is minimal compared to the lookup time of each group
+            const int maxParams = 2000;
+            if (ids.Length <= maxParams)

```

Comment:
There's actually no reason to have this check, it's easy to just always do InGroupsOf 2000 regardless of how many

Reply:
@Shazwazza it's easier yes - but you'll also introduce the `.AddRange()` overhead to each and every query.